pkg/compressio/nocompressio.go

```diff
@@ -0,0 +1,284 @@
+// |                  data size (4-bytes)                 |
+// +------------------------------------------------------+
+// |                  data                                |
+// +------------------------------------------------------+
+// |       (optional) hash (32-bytes)                     |
+// +------------------------------------------------------+
+// |                  data size (4-bytes)                 |
+// +------------------------------------------------------+
+// |                       ......                         |
+// \------------------------------------------------------/
+//
+// where each hash is calculated from the following items in order
+//
+//	data
+//	data size
+
+// SimpleReader is a reader from uncompressed image.
+type SimpleReader struct {
+	// in is the source.
+	in io.Reader
+
+	// key is the key used to create hash objects.
+	key []byte
+
+	// h is the hash object.
+	h hash.Hash
+
+	// current data chunk size
+	chunkSize uint32
+
+	// current chunk position
+	done uint32
+}
+
+var _ io.Reader = (*SimpleReader)(nil)
+
+const (
+	defaultBufSize = 256 * 1024
+)
+
+// NewSimpleReader returns a new (uncompressed) reader. If key is non-nil, the data stream
+// is assumed to contain expected hash values. See package comments for
+// details.
+func NewSimpleReader(in io.Reader, key []byte) (*SimpleReader, error) {
+	r := &SimpleReader{
+		in:  bufio.NewReaderSize(in, defaultBufSize),
+		key: key,
+	}
+
+	if key != nil {
+		r.h = hmac.New(sha256.New, key)
+	}
+
+	return r, nil
+}
+
+// ReadByte implements wire.Reader.ReadByte.
+func (r *SimpleReader) ReadByte() (byte, error) {
+	var p [1]byte
+	n, err := r.Read(p[:])
+	if n != 1 {
+		return p[0], err
+	}
+	// Suppress EOF.
+	return p[0], nil
+}
+
+// Read implements io.Reader.Read.
+func (r *SimpleReader) Read(p []byte) (int, error) {
+	var scratch [4]byte
+
+	if len(p) == 0 {
+		return r.in.Read(p)
+	}
+
+	// need next chunk?
+	if r.done >= r.chunkSize {
+		if _, err := io.ReadFull(r.in, scratch[:]); err != nil {
+			return 0, err
+		}
+
+		r.chunkSize = binary.BigEndian.Uint32(scratch[:])
+		r.done = 0
+		if r.key != nil {
+			r.h.Reset()
+		}
+
+		if r.chunkSize == 0 {
+			// this must not happen
+			return 0, io.ErrNoProgress
+		}
+	}
+
+	toRead := uint32(len(p))
+	// can't read more than whats left
+	if toRead > r.chunkSize-r.done {
+		toRead = r.chunkSize - r.done
+	}
+
+	n, err := r.in.Read(p[:toRead])
+	if err != nil {
+		if err == io.EOF {
+			// this only can happen if storage or data size is corrupted,
+			// but we have no other means to detect it earlier as we store
+			// hash after the data block.
+			return n, ErrHashMismatch
+		}
+		return n, err
+	}
+
+	if r.key != nil {
+		_, _ = r.h.Write(p[:n])
+	}
+
+	r.done += uint32(n)
+	if r.done >= r.chunkSize {
+		if r.key != nil {
+			binary.BigEndian.PutUint32(scratch[:], r.chunkSize)
+			r.h.Write(scratch[:4])
+
+			sum := r.h.Sum(nil)
+
+			var readerSum []byte = make([]byte, len(sum))
```

Comment:
Sorry, yet another stubborn internal-only test is failing due to this:
```
// pkg/compressio/nocompressio.go:155: should omit type []byte from declaration; it will be inferred from the right-hand side
```

Can you remove the `[]byte` type annotation?